### ISSUE TITLE
Validation array with model

### ISSUE THREAD

**Issue by pocman** [orgsync/active_interaction#301]
Using 1.6, I'm not able to validate
```
array :contracts
```
or 
```
array :contracts do
   model
end
```
or
```
array :contracts do
   model class: Contract
end
```

**Comment by tfausak** [comment]
What kind of error are you getting? I can successfully run this code on the `v1.6.0` tag.

``` rb
class Contract
end

class Example < ActiveInteraction::Base
  array :contracts

  def execute
    inputs
  end
end

Example.run!(contracts: [Contract.new])
# => {:contracts=>[#<Contract:0x007fe0dba25468>]}
```

**Comment by pocman** [comment]
My issue came from the fact that I was not passing an array but an
active_query.
how can I validate ?

`Example.run!(contracts: Contract.all)`

Thanks for your help.

2015-09-30 17:07 GMT+02:00 Taylor Fausak <notifications@github.com>:

> What kind of error are you getting? I can successfully run this code on
> the v1.6.0 tag.
>
> class Contractend
> class Example < ActiveInteraction::Base
>   array :contracts
>
>   def execute
>     inputs
>   endend
> Example.run!(contracts: [Contract.new])# => {:contracts=>[#<Contract:0x007fe0dba25468>]}
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/orgsync/active_interaction/issues/301#issuecomment-144440415>
> .
>



-- 
*Thomas Pocreau*
Recherche & Développement | *Colisweb*

thomas@colisweb.com
06 45 12 61 58
Euratechnologies | 165, Avenue de Bretagne
59000 - Lille

**Comment by tfausak** [comment]
Array input should accept `ActiveRecord::Relation` objects. We added that in v1.4 with #239.

**Comment by pocman** [comment]
I have this error : 
```
ActiveInteraction::InvalidInteractionError: Contracts translation missing: fr.active_interaction.errors.models.cheapest_stores_contract_for_dimensions_and_shipping_zip_code.attributes.cntracts.invalid_type
        from .../vendor/bundle/ruby/2.1.0/gems/active_interaction-1.6.0/lib/active_interaction/concerns/runnable.rb:111:in `run!'
        from .../vendor/bundle/ruby/2.1.0/gems/active_interaction-1.6.0/lib/active_interaction/concerns/runnable.rb:141:in `run!'
        from (irb):5
        from .../vendor/bundle/ruby/2.1.0/gems/railties-3.2.22/lib/rails/commands/console.rb:47:in `start'
        from .../vendor/bundle/ruby/2.1.0/gems/railties-3.2.22/lib/rails/commands/console.rb:8:in `start'
        from .../vendor/bundle/ruby/2.1.0/gems/railties-3.2.22/lib/rails/commands.rb:41:in `<top (required)>'
        from script/rails:6:in `require'
        from script/rails:6:in `<main>'
```

where `Store.last.contracts.class => Array`

**Comment by tfausak** [comment]
You may have a typo somewhere. That error mentions `cntracts` without an "o".

**Comment by pocman** [comment]
It's just a typo when I formatted the error message. (Sorry about that).

**Comment by tfausak** [comment]
Which version of ActiveRecord are you using? And which database driver? I cannot replicate this behavior. Here is the file I am using to test:

``` rb
require 'active_record' # 4.2.4
require 'active_interaction' # 1.6.0

ActiveRecord::Base.establish_connection(
  adapter: 'sqlite3', # 1.3.10
  database: ':memory:'
)

ActiveRecord::Schema.define do
  create_table :stores

  create_table :contracts do |table|
    table.column :store_id, :integer
  end
end

class Store < ActiveRecord::Base
  has_many :contracts
end

class Contract < ActiveRecord::Base
  belongs_to :store
end

class Example < ActiveInteraction::Base
  array :contracts

  def execute
    inputs
  end
end

store = Store.create!
contract = store.contracts.create!

Example.run!(contracts: store.contracts)
# => {:contracts=>#<ActiveRecord::Associations::CollectionProxy [#<Contract id: 1, store_id: 1>]>}
Example.run!(contracts: Contract.all)
# => {:contracts=>#<ActiveRecord::Relation [#<Contract id: 1, store_id: 1>]>}
Example.run!(contracts: [contract])
# => {:contracts=>[#<Contract id: 1, store_id: 1>]}
```

**Comment by pocman** [comment]
We are using activerecord (3.2.22)

**Comment by tfausak** [comment]
Aha! I can replicate the bad behavior with ActiveRecord 3.2.22. It only affects associations, not relations. So `store.contracts` doesn't work, but `Contract.scoped` does.

**Comment by tfausak** [comment]
This problem still occurs on master (v2.1.2).

**Comment by tfausak** [comment]
It does not affect ActiveRecord 4.0.0. It does affect 3.2.0. I think it's safe to assume all 3.2.x versions of ActiveRecord are affected.

**Comment by tfausak** [comment]
The fix has been released in version 2.1.3. Do you need this change back ported to 1.6? If so, what's keeping you from upgrading to version 2?

**Comment by pocman** [comment]
Yes please.
We are not ready to migrate yet!

Thanks for you're reactivity.

**Comment by tfausak** [comment]
Ok. I have released 1.6.1 with this bug fix. Thanks for reporting it!
